### PR TITLE
[TF FE]: Support complex tensors for Sub operation

### PR DESCRIPTION
**Pull request by hub-bla**
### Details:
 - Support complex tensors for Sub operation + tests

### Tickets:
 - [None](https://github.com/openvinotoolkit/openvino/issues/22948)

### CODE REVIEW

File: src/frontends/tensorflow/src/op_table.cpp [hunk 1]
```diff
@@ -195,7 +195,6 @@ const std::map<std::string, CreatorFunction> get_supported_ops() {
         {"Pow", CreatorFunction(translate_binary_op<v1::Power>)},
         {"RealDiv", CreatorFunction(translate_binary_op<v1::Divide>)},
         {"SquaredDifference", CreatorFunction(translate_binary_op<v0::SquaredDifference>)},
-        {"Sub", CreatorFunction(translate_binary_op<v1::Subtract>)},
```
Comment (rkazants) [inline src/frontends/tensorflow/src/op_table.cpp:198]: please remove `#include "openvino/op/subtract.hpp"`. It looks no longer needed

File: src/frontends/tensorflow_common/src/op/binary_op.cpp [hunk 2]
```diff
@@ -171,6 +171,33 @@ OutputVector translate_addv2_op(const NodeContext& node) {
...
+        auto complex_result = make_shared<ComplexTypeMark>(result, complex_type_mark_lhs->get_complex_part_type());
+        set_node_name(node.get_name(), result);
+
+        return {complex_result};
+    }
+    set_node_name(node.get_name(), result);
+    return {result};
+}
+
 template OutputVector translate_binary_op<v1::Add>(const NodeContext& node);
```
Comment (rkazants) [inline src/frontends/tensorflow_common/src/op/binary_op.cpp:201]: please remove `template OutputVector translate_binary_op<v13::BitwiseAnd>(const NodeContext& node);` and `template OutputVector translate_binary_op<v1::Subtract>(const NodeContext& node);`
They look no longer needed.

File: tests/layer_tests/tensorflow_tests/test_tf_Sub.py [hunk 3]
```diff
@@ -216,3 +216,70 @@ def test_sub_placeholder_const_broadcast_5D(self, params, ie_device, precision,
...
+            param_real_x = tf.compat.v1.placeholder(np.float32, x_shape, 'param_real_x')
+            param_imag_x = tf.compat.v1.placeholder(np.float32, x_shape, 'param_imag_x')
+            
+            param_real_y = tf.compat.v1.placeholder(np.float32, y_shape, 'param_real_y')
+            param_imag_y = tf.compat.v1.placeholder(np.float32, y_shape, 'param_imag_y')
+
+            x = tf.raw_ops.Complex(real=param_real_x, imag=param_imag_x)
+            y = tf.raw_ops.Complex(real=param_real_y, imag=param_imag_y)
+
+            result = tf.subtract(x, y, name="Operation")
```
Comment (rkazants) [inline tests/layer_tests/tensorflow_tests/test_tf_Sub.py:260]: please use `tf.raw_ops.Sub` instead. `tf.subtract` may be a wrapper

**Review by rkazants — APPROVED** [review]
Left some comments, please address. The remaining part looks good to me

File: tests/layer_tests/tensorflow_tests/test_tf_Sub.py (hunk 3 above)
Comment (rkazants) [inline tests/layer_tests/tensorflow_tests/test_tf_Sub.py:260]:
```suggestion
            result = tf.raw_ops.Sub(x=x, y=y, name='Sub')
```

**Comment by rkazants** [comment]
build_jenkins

File: src/frontends/tensorflow_common/src/op/binary_op.cpp (hunk 2 above)
Comment (hub-bla) [reply src/frontends/tensorflow_common/src/op/binary_op.cpp:201]: after deleting these lines i get
`
openvino/bin/intel64/Release/libopenvino_tensorflow_lite_frontend.so.2024.4.0: undefined reference to std::vector<ov::Output<ov::Node>, std::allocator<ov::Output<ov::Node> > > ov::frontend::tensorflow::op::translate_binary_op<ov::op::v1::Subtract>(ov::frontend::NodeContext const&)
`

Comment (hub-bla) [reply src/frontends/tensorflow_common/src/op/binary_op.cpp:201]: Also, it seems like `BitwiseAnd` in `op_table.cpp` still looks like this `{"BitwiseAnd", CreatorFunction(translate_binary_op<v13::BitwiseAnd>)},`. 

Did you meant to delete template `OutputVector translate_binary_op<v1::Add>(const NodeContext& node);` instead?

Comment (hub-bla) [reply src/frontends/tensorflow_common/src/op/binary_op.cpp:201]: https://github.com/openvinotoolkit/openvino/blob/04e0fea532159d0993b0af5747703db8c4825ffe/src/frontends/tensorflow_lite/src/op/op_translation_utils.hpp#L50-L62

https://github.com/openvinotoolkit/openvino/blob/04e0fea532159d0993b0af5747703db8c4825ffe/src/frontends/tensorflow/src/op_table.cpp#L198

I assume this part in tensorflow_lite might be the reason for the error. Should I update Sub there?

Comment (rkazants) [reply src/frontends/tensorflow_common/src/op/binary_op.cpp:201]: Yes, you look right. So let us leave it as is

**Review by rkazants — APPROVED** [review]

**Comment by rkazants** [comment]
build_jenkins

**Pull request merged**